src/s3fs.cpp

```diff
@@ -2708,6 +2721,472 @@ static int remote_mountpath_exists(const char* path)
+  string      strpath;
+  string      newpath;
+  string      nowcache;
+  headers_t   meta;
+  struct stat stbuf;
+  int         nDirType = DIRTYPE_UNKNOWN;
+
+  if(0 == strcmp(path, "/")){
+    DPRNNN("Could not change mode for mount point.");
+    return -EIO;
+  }
+  if(0 != (result = check_parent_object_access(path, X_OK))){
+    return result;
+  }
+  if(0 != (result = check_object_owner(path, &stbuf))){
+    return result;
+  }
+
+  if(S_ISDIR(stbuf.st_mode)){
+    result = chk_dir_object_type(path, newpath, strpath, nowcache, &meta, &nDirType);
+  }else{
+    strpath  = path;
+    nowcache = strpath;
+    result   = get_object_attribute(strpath.c_str(), NULL, &meta);
+  }
+  if(0 != result){
+    return result;
+  }
+
+  // make new header_t
+  if(0 != (result = set_xattrs_to_header(meta, name, value, size, flags))){
+    return result;
+  }
+
+  if(S_ISDIR(stbuf.st_mode) && IS_REPLACEDIR(nDirType)){
+    // Should rebuild directory object(except new type)
+    // Need to remove old dir("dir" etc) and make new dir("dir/")
+
+    // At first, remove directory old object
+    if(IS_RMTYPEDIR(nDirType)){
+      S3fsCurl s3fscurl;
+      if(0 != (result = s3fscurl.DeleteRequest(strpath.c_str()))){
+        return result;
+      }
+    }
+    StatCache::getStatCacheData()->DelStat(nowcache);
+
+    // Make new directory object("dir/")
+    if(0 != (result = create_directory_object(newpath.c_str(), stbuf.st_mode, stbuf.st_mtime, stbuf.st_uid, stbuf.st_gid))){
+      return result;
+    }
+
+    // need to set xattr header for directory.
+    strpath  = newpath;
+    nowcache = strpath;
+  }
+
+  // set xattr all object
+  meta["x-amz-copy-source"]        = urlEncode(service_path + bucket + get_realpath(strpath.c_str()));
+  meta["x-amz-metadata-directive"] = "REPLACE";
+
+  if(0 != put_headers(strpath.c_str(), meta, true)){
+    return -EIO;
+  }
+  StatCache::getStatCacheData()->DelStat(nowcache);
+
+  return 0;
+}
+
+static int s3fs_getxattr(const char* path, const char* name, char* value, size_t size)
+{
+  FPRN("[path=%s][name=%s][value=%p][size=%zu]", path, name, value, size);
+
+  if(!path || !name){
+    return -EIO;
+  }
+
+  int       result;
+  headers_t meta;
+  xattrs_t  xattrs;
+
+  // check parent directory attribute.
+  if(0 != (result = check_parent_object_access(path, X_OK))){
+    return result;
+  }
+
+  // get headders
+  if(0 != (result = get_object_attribute(path, NULL, &meta))){
+    return result;
+  }
+
+  // get xattrs
+  if(meta.end() == meta.find("x-amz-meta-xattr")){
+    // object does not have xattrs
+    return -ENOATTR;
+  }
+  string strxattrs = meta["x-amz-meta-xattr"];
+
+  parse_xattrs(strxattrs, xattrs);
+
+  // search name
+  string strname = name;
+  if(xattrs.end() == xattrs.find(strname)){
+    // not found name in xattrs
+    free_xattrs(xattrs);
+    return -ENOATTR;
+  }
+
+  // decode
+  size_t         length = 0;
+  unsigned char* pvalue = NULL;
+  if(NULL != xattrs[strname]){
```

Comment:
This should always be true -- `xattrs[strname]` will call the default constructor here.